tests/core/celery/tasks/test_generate_inventory_and_hold_reports.py

```diff
@@ -0,0 +1,279 @@
+    inventory_row_count = 0
+
+    for row in inventory_report_reader:
+        inventory_row_count += 1
+        assert row["title"] == title
+        assert row["author"] == author
+        assert row["identifier"]
+        assert row["language"] == language
+        assert row["publisher"] == publisher
+        assert row["audience"] == "young adult"
+        assert row["genres"] == "genre_a,genre_z"
+        assert row["format"] == edition.BOOK_MEDIUM
+        assert row["data_source"] == data_source
+        assert row["collection_name"] == collection_name
+        assert float(row["days_remaining_on_license"]) == float(days_remaining)
+        assert row["shared_active_loan_count"] == "0"
+        assert row["library_active_loan_count"] == "0"
+        assert row["remaining_loans"] == str(checkouts_left)
+        assert row["allowed_concurrent_users"] == str(terms_concurrency)
+        assert expiration.strftime("%Y-%m-%d %H:%M:%S.%f") in row["license_expiration"]
+
+    assert inventory_row_count == 1
+
+    holds_report_key = [x for x in attachments.keys() if "holds" in x][0]
+    assert holds_report_key
+    assert "test_library" in holds_report_key
+    holds_report_value = attachments[holds_report_key]
+    assert holds_report_value
+    holds_report_reader = csv.DictReader(open(holds_report_value))
+    row_count = 0
+
+    for row in holds_report_reader:
+        row_count += 1
+        assert row["title"] == title
+        assert row["author"] == author
+        assert row["identifier"]
+        assert row["language"] == language
+        assert row["publisher"] == publisher
+        assert row["audience"] == "young adult"
+        assert row["genres"] == "genre_a,genre_z"
+        assert row["format"] == edition.BOOK_MEDIUM
+        assert row["data_source"] == data_source
+        assert row["collection_name"] == collection_name
+        assert int(row["shared_active_hold_count"]) == shared_patrons_in_hold_queue
+        assert int(row["library_active_hold_count"]) == 3
+
+        assert row_count == 1
+
+        # clean up files
+        for f in attachments.values():
+            os.remove(f)
+
+
+def create_test_opds_collection(collection_name, data_source, db, library):
+    settings = OPDSImporterSettings(
+        include_in_inventory_report=True,
+        external_account_id="http://opds.com",
+        data_source=data_source,
+    )
+    collection = db.collection(name=collection_name, settings=settings.dict())
+    collection.libraries = [library]
+    return collection
+
+
+def test_generate_inventory_and_hold_reports_task(
+    db: DatabaseTransactionFixture, celery_fixture: CeleryFixture
+):
+    library = db.library(short_name="test_library")
+    # there must be at least one opds collection associated with the library for this to work
+    create_test_opds_collection("c1", "d1", db, library)
+    GenerateInventoryAndHoldsReportsJob._delete_attachments = True
+    services = container_instance()
```

Comment:
Instead of directly calling `container_instance` it would be better to use the `ServicesFixture` pytest fixture. 

Then you could override `send_email` like this:
```python
def test_generate_inventory_and_hold_reports_task(
    db: DatabaseTransactionFixture, celery_fixture: CeleryFixture, services_fixture: ServicesFixture
):
   ...
   send_email_mock = create_autospec(services_fixture.services.email.send_email)
   services_fixture.services.email.send_email.override(send_email_mock)
```